docs/rfc/0698-cli-infrastructure.adoc

```diff
@@ -0,0 +1,355 @@
+  modify and retrieve `Project` identities.
+* `person [ create | get | list | update | checkout | review ]` --
+  modify and retrieve `Person` identities.
+* `any [ get | list ]` -- retrieve identities where the caller does not
+  know what kind of identity it is. The CLI will make a best-effort to
+  infer the type, or else return an opaque blob.
+* `local [ set | get | default ]` -- modify or retrieve a local identity
+* `tracking [ track | untrack | list ]` -- modify and list the tracking
+  graph for a given identity.
+* `rad-refs [ self | signed | delegates | delegate ]` -- retrieve the
+  blobs that live under `rad/<refname>`
+* `refs [ heads | notes | tags | category ]` -- list the reference names.
+  under the respective category, where `category` is a user supplied category.
+
+=== `rad-replicate`
+
+* `ask` -- tell the `daemon` that you wish to replicate the
+  given URN. Pending <<rl-141, radicle-link#141>>.
+* `status` -- retrieve the status of a URN that we have `ask`ed for.
+* `list` -- list the URNs that we have asked for but have not yet replicated.
+* `direct` -- if the peer/address are known for the given URN,
+  attempt to replicate the identity. This requires the activation of
+  the peer-to-peer daemon.
+
+=== `rad-profile`
+
+This command will provide the management of keys and the `Profile`
+construct in `librad`. The following actions will be defined:
+
+* `create`— generate a new `Profile`-- setting up a new key and any
+  related infrastructure for the profile, e.g. storage.
+* `get`— get the current `Profile`'s identifier.
+* `set`— set the current `Profile` to the identifier.
+  passed in.
+* `list`— list the available `Profile` identifiers.
+* `ssh-add`— add the current `Profile`'s key to the `ssh-agent`.
+
+=== `rad-daemon`
+
+This command provides introspection into the peer-to-peer daemon.
+
+* `peers` -- get the list of currently connected peers
+* `membership` -- get the membership list for the peer-to-peer
+  daemon's membership view.
+* `stats` -- request the stats of the peer-to-peer daemon, i.e. a
+  summary of peers and members.
+
+=== Note on Output
+
+Some operations will result in data that should be output, for
+example, retrieving a `Project` from the store will return the
+metadata for that project. As per <<arch-cli, RFC-682#CLI>>:
+
+[quote]
+-------
+
+Each subcommand MUST expose its functionality as a linkable library,
+and provide CBOR <<RFC8949>> serialisation for its arguments and outputs.
+-------
+
+The output should, however, not be limited to CBOR. We will provide an
+option, `--output`, that will give the caller the option to output
+into another format. The initial options will be:
+
+. `json`
+. `cbor`
+
+with the default being `json` since the CLIs are user-centric
+applications and human-readable footnote:disclaimer[For some
+definition of human-readble] output would be desirable. More output
+types MAY be proposed in future RFCs.
+
+== Documentation
```

Comment:
Thanks for taking the time to research this. I get that the outcome is that there is no tooling for such a thing in Rust. If it is necessary to standardise on the option parsing library (because that is part of the subcommand library surface), then I'd be in favour of `clap`.

Otherwise, it seems like documentation equals hand curated `man` pages, like in the olden days.

Reply:
There's no such tooling, but there's some groundwork at least for `clap`. I'm also in favour of it.

> Otherwise, it seems like documentation equals hand curated man pages, like in the olden days.
Also got you covered :upside_down_face: https://github.com/rust-cli/man

Reply:
Hm, TIL a clap “app” can also be generated from YAML, so maybe a unison program which outputs dhall which outputs yaml could be a solution 🤔

Reply:
You only get one innovation point per project :stuck_out_tongue_closed_eyes:

Reply:
NoveltyCoin (ℕ)